src/process.rs

```diff
@@ -424,12 +475,45 @@ mod tests {
 
         // start_year after end_year
         assert!(
-            create_param_raw(Some(2001), Some(2000), Some(1.0), Some(0.0))
+            create_param_raw(Some(2001), Some(2000), 1, Some(1.0), Some(0.0))
                 .into_parameter(&year_range)
                 .is_ok()
         );
     }
 
+    #[test]
+    #[should_panic]
+    fn test_param_raw_validate_bad_lifetime() {
+        // lifetime = 0
+        assert!(
+            create_param_raw(Some(2000), Some(2100), 0, Some(1.0), Some(0.0))
+                .validate()
+                .is_ok()
+        );
+    }
```

Comment:
This would be clearer:

```suggestion
    #[test]
    fn test_param_raw_validate_bad_lifetime() {
        // lifetime = 0
        assert!(
            create_param_raw(Some(2000), Some(2100), 0, Some(1.0), Some(0.0))
                .validate()
                .is_err()
        );
    }
```

No need to check that the test panics. (I just noticed that I've done the same thing in the test above for some reason!)

Same thing goes for the tests below.

Reply:
I was just copying your tests and replacing them with my stuff :P